src/framing/Frame.cpp

```diff
@@ -94,78 +94,135 @@ std::ostream& operator<<(std::ostream& os, ErrorCode errorCode) {
   return os << "ErrorCode(" << static_cast<uint32_t>(errorCode) << ")";
 }
 
-std::ostream& operator<<(std::ostream& os, FrameFlags frameFlags) {
-  // TODO Match the Flag names with the AllowedFlags in the Frame declarations
+std::ostream& operator<<(
+    std::ostream& os,
+    const std::pair<FrameFlags, FrameFlags>& frameFlagsAndAllowedFlags) {
+  FrameFlags frameFlags = frameFlagsAndAllowedFlags.first;
+  FrameFlags allowedFlags = frameFlagsAndAllowedFlags.second;
+  FrameFlags foundFlags = FrameFlags::EMPTY;
 
   std::stringstream ss;
   std::string delimeter = "";
-  if (!!(frameFlags & FrameFlags::NEXT)) {
-    ss << "NEXT";
+  if (!!(allowedFlags & FrameFlags::IGNORE)
```

Comment:
Would it be possible to loop over these flags instead of all the if statements?
You can use the to_string function to print out the flags.